src/mlpack/methods/reinforcement_learning/async_learning.hpp

```diff
@@ -0,0 +1,144 @@
+namespace rl {
+
+/**
+ * Wrapper of various asynchronous learning algorithms,
+ * e.g. async one-step Q-learning, async one-step Sarsa,
+ * async n-step Q-learning and async advantage actor-critic.
+ *
+ * For more details, see the following:
+ * @code
+ * @inproceedings{mnih2016asynchronous,
+ *   title     = {Asynchronous methods for deep reinforcement learning},
+ *   author    = {Mnih, Volodymyr and Badia, Adria Puigdomenech and Mirza,
+ *                Mehdi and Graves, Alex and Lillicrap, Timothy and Harley,
+ *                Tim and Silver, David and Kavukcuoglu, Koray},
+ *   booktitle = {International Conference on Machine Learning},
+ *   pages     = {1928--1937},
+ *   year      = {2016}
+ * }
+ * @endcode
+ *
+ * @tparam WorkerType The type of the worker.
+ * @tparam EnvironmentType The type of reinforcement learning task.
+ * @tparam NetworkType The type of the network model.
+ * @tparam UpdaterType The type of the optimizer.
+ * @tparam PolicyType The type of the behavior policy.
+ */
+template <
+  typename WorkerType,
+  typename EnvironmentType,
+  typename NetworkType,
+  typename UpdaterType,
+  typename PolicyType
+>
+class AsyncLearning
+{
+ public:
+  /**
+   * Construct an instance of the given async learning algorithm.
+   *
+   * @param config Hyper-parameters for training.
+   * @param network The network model.
+   * @param policy The behavior policy.
+   * @param updater The optimizer.
+   * @param environment The reinforcement learning task.
+   */
+  AsyncLearning(TrainingConfig config,
+                NetworkType network,
+                PolicyType policy,
+                UpdaterType updater = UpdaterType(),
+                EnvironmentType environment = EnvironmentType());
+
+  /**
+   * Starting async training.
+   *
+   * @tparam Measure The type of the measurement. It should be a
+   *   callable object like
+   *   @code
+   *   bool foo(double reward);
+   *   @endcode
+   *   where reward is the total reward of a deterministic test episode,
+   *   and the return value should indicate whether the training
+   *   process is completed.
+   * @param measure The measurement instance.
+   */
+  template <typename Measure>
+  void Train(Measure& measure);
+
+  //! Get and modify training config.
+  TrainingConfig& Config() { return config; }
+  const TrainingConfig& Config() const { return config; }
+
+  //! Get and modify learning network.
+  NetworkType& Network() { return learningNetwork; }
+  const NetworkType& Network() const { return learningNetwork; }
+
+  //! Get and modify behavior policy.
+  PolicyType& Policy() { return policy; }
+  const PolicyType& Policy() const { return policy; }
+
+  //! Get and modify optimizer.
+  UpdaterType& Updater() { return updater; }
+  const UpdaterType& Updater() const { return updater; }
+
+  //! Get and modify the environment.
+  EnvironmentType& Environment() { return environment; }
+  const EnvironmentType& Environment() const { return environment; }
```

Comment:
Very minor doxygen issue---can you make one comment for each of these methods instead of a comment that covers both?  Doxygen won't pick up any documentation for `const EnvironmentType& Environment() const` like this is now.